### ISSUE TITLE
Compaction after many appends (using 'overwrite' mode?)

### ISSUE THREAD

**Issue by erinov1** [dask/fastparquet#733]
I was unable to find any outstanding Issues surrounding compaction, even though it seems like a common use-case. Suppose I have a partitioned parquet dataset `my_dataset`, partitioned by year. Initially,
```
my_dataset
├── _common_metadata
├── _metadata
├── partition=2021
│   └── part.0.parquet
├── partition=2022
    └── part.0.parquet
```
Next, once a day I `write` new data to `my_dataset` using `append=True` mode; since the year is currently 2022, the new data will belong to `partition=2022`. After several days:
```
my_dataset
├── _common_metadata
├── _metadata
├── partition=2021
│   └── part.0.parquet
├── partition=2022
    ├── part.0.parquet
    ├── part.1.parquet
    ├── part.2.parquet
    └── part.3.parquet
```
Every so often I would like to compact the files in `partition=2022` into a single file. Is there a way to do this with existing `fastparquet` functionality? 

A crude attempt would be to load up the all data in `partition=2022` into `pandas`, then write back to the dataset with `append=overwrite`. However, this does not work when a partition contains multiple files (as in this example) — see the discussion in https://github.com/dask/fastparquet/pull/712. Is it instead possible to load up the data in `partition=2022`, do an ordinary `append=True`, and then modify the metadata to ignore the old files? Presumably one could also get a list of files that are "active" from the metadata (and then manually delete any "inactive" files if one wishes). Or is there a better solution?

**Comment by yohplala** [comment]
Hi @erinov1 ,

> A crude attempt would be to load up the all data in `partition=2022` into pandas, then write back to the dataset with `append=overwrite`.

This is actually the target use case for `overwrite`.
In #712 has been on the contrary removed this limit.
Before #712, a unique parquet file per folder was a constraint (the advantage is that it was not necessary to re-order the list of parquet files - the file being written was taking the same position than the file that was being suppressed)

With #712, row groups are now re-ordered per partition.
This re-ordering step allows for adding as many files as you want, in the order you want.
Bests

PS: yes, this might be 'crude', yet `overwrite` is the most versatile option. Yes you need to load in pandas your data, but you can do then all the data processing you wish (removing duplicates the way you want, or not removing them, and so on...) This 'processing' step is too-much user dependent. I have not seen any 'simple' yet 'versatile' method to implement in fastparquet.
(I have implemented one, for ordered dataset, not partition based, in [oups](https://github.com/yohplala/oups), but I am relying on several assumptions so that it makes sense)

**Comment by erinov1** [comment]
Thanks @yohplala ! I misunderstood https://github.com/dask/fastparquet/pull/712. I was also using the latest tagged version `0.7.2` which does not contain the changes. 

If I understand correctly, old row groups in a given partition appear to be removed: https://github.com/dask/fastparquet/blob/263df0aac519f94554a8ec869711015f3bb0f358/fastparquet/writer.py#L1460-L1461 

before the new data is written: https://github.com/dask/fastparquet/blob/263df0aac519f94554a8ec869711015f3bb0f358/fastparquet/writer.py#L1468-L1471

This could be a concern if the latter writing step fails, in which case all the data would be lost. I wonder if it is possible to add functionality to keep the old files in such a way that they can either be deleted once no longer needed (either through an API or manually), or restored to the dataset in case something goes awry.

**Comment by martindurant** [comment]
Agreed that deleting only after successful append of combined row group makes sense. I assume this would mean simply reversing the order of the two steps. You should note, though @erinov1 , that writing the new metadata always comes with a chance of destruction/corruption, should (for example) the process be suddenly terminated.

**Comment by yohplala** [comment]
Hi @erinov1 , hi @martindurant 

Thanks @erinov1 for the feedback, I created PR #734 that takes it into account.
It was not as simple as _simply reversing the order of the two steps_ @martindurant , but clearly not infeasible either :).
I took the opportunity to rename the function into `overwrite` which I was procrastinating to do. This might be the right time as it would seem this feature is finding a public :).
